tools/load_balancing_tool/tests/load_balancing_test.py

```diff
@@ -0,0 +1,288 @@
+            jsonfile1.flush()
+            cmd = "./load_balancing_solve.py --json_input %s --json_output %s" % (jsonfile1.name, jsonfile2.name)
+            output = run_cmd_no_fail(cmd, from_dir=CODE_DIR)
+            self._check_solution(output, "NTASKS_ATM", 992)
+            cmd = "./load_balancing_solve.py --json_input %s" % jsonfile2.name
+            output = run_cmd_no_fail(cmd, from_dir=CODE_DIR)
+            self._check_solution(output, "NTASKS_ATM", 992)
+
+
+    def test_solve_from_timing_dir(self):
+        cmd = "./load_balancing_solve.py --timing_dir %s --total_tasks 64 --blocksize 2" % os.path.join(TEST_DIR, "timing")
+        output = run_cmd_no_fail(cmd, from_dir=CODE_DIR)
+        self._check_solution(output, "NTASKS_ATM", 62)
+
+    def test_write_pes(self):
+        with tempfile.NamedTemporaryFile('w+') as jsonfile1, tempfile.NamedTemporaryFile('w+') as pes_file:
+            json.dump(JSON_DICT, jsonfile1)
+            jsonfile1.flush()
+            cmd = "./load_balancing_solve.py --json_input %s --pe_output %s" % (jsonfile1.name, pes_file.name)
+            output = run_cmd_no_fail(cmd, from_dir=CODE_DIR)
+            
+            self.assertTrue(os.access(pes_file.name, os.R_OK), "pesfile %s not written" % pes_file.name)
+            pesobj = CIME.XML.pes.Pes(pes_file.name)
+        for node in pesobj.get_nodes('pes'):
+            pesize = node.get('pesize')
+            pes_ntasks, pes_nthrds, pes_rootpe, ignore = \
+               pesobj.find_pes_layout('any', 'any', 'any', pesize_opts=pesize)
+
+    def test_set_blocksize_atm(self):
+        cmd = "./load_balancing_solve.py --timing_dir %s --total_tasks 64 --blocksize 2 --blocksize_atm 4" % os.path.join(TEST_DIR, "timing")
+        output = run_cmd_no_fail(cmd, from_dir=CODE_DIR)
+        self._check_solution(output, "NTASKS_ATM", 60)
+        self._check_solution(output, "NBLOCKS_ATM", 15)
+        self._check_solution(output, "NTASKS_OCN", 4)
+        self._check_solution(output, "NBLOCKS_OCN", 2)
+
+    def test_graph_models(self):
+        try:
+            import matplotlib
+        except ImportError, e:
+            self.skipTest("matplotlib not found")
+
+        with tempfile.NamedTemporaryFile('w+') as jsonfile:
+            json.dump(JSON_DICT, jsonfile)
+            jsonfile.flush()
+            cmd = "./load_balancing_solve.py --json_input %s --graph_models" % (jsonfile.name)
+            output = run_cmd_no_fail(cmd, from_dir=CODE_DIR)
+            self._check_solution(output, "NTASKS_ATM", 992)
+        
+    def test_xcase_submit(self):
+        with tempfile.NamedTemporaryFile('w+') as tfile, tempfile.NamedTemporaryFile('w+') as xfile:
+            tfile.write(PES_XML)
+            tfile.flush()
+            xfile.write(X_OPTIONS)
+            xfile.flush()
+            cmd = "./load_balancing_submit.py --pesfile %s --res ne30_g16 --compset X --casename_prefix test_lbt_  --extra_options_file %s" % (tfile.name, xfile.name)
```

Comment:
I don't think that this test should run by default under the scripts directory.   This directory may not be writable by the user.  This is a moot point if we choose to use the PFS test instead of create_newcase.